components/ProtectedContentBundle/bundle/Command/CleanTokenCommand.php

```diff
@@ -14,25 +14,24 @@
 
 namespace Novactive\Bundle\eZProtectedContentBundle\Command;
 
-use Doctrine\ORM\EntityManagerInterface;
 use Novactive\Bundle\eZProtectedContentBundle\Entity\ProtectedTokenStorage;
 use Novactive\Bundle\eZProtectedContentBundle\Repository\ProtectedTokenStorageRepository;
 use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 class CleanTokenCommand extends Command
 {
     /**
-     * @var EntityManagerInterface
+     * @param ContainerInterface $container
      */
-    private $entityManager;
 
-    public function __construct(EntityManagerInterface $entityManager)
+    public function __construct(ContainerInterface $container)
```

Comment:
Injecting container is considered a bad practice.
What is the problem with EntityManagerInterface type hinting ?

Reply:
PR Draft sur le bundle : https://github.com/Novactive/Nova-eZPlatform-Bundles/pull/180 (Qui sert juste à illustrer le pb)

Reply:
Le problème rencontré vient du fait que le service factorisé (novaezprotectedcontent.doctrine.entity_manager) est injecté par défaut dans tous les services. La commande fonctionne correctement lorsque j'injecte @doctrine.orm.entity_manager au lieu de celui par défaut. Je pense que c'est une solution optimale plutôt que d'injecter tout le container.

Reply:
le service `novaezprotectedcontent.doctrine.entity_manager` avait été créé afin d'avoir un entity manager qui était siteaccess aware.
Dans la version 4 de ibexa, il existe un entity manager ibexa rempli ce rôle et qui selon peut remplacer dans problème le service `novaezprotectedcontent.doctrine.entity_manager`